src/java/org/apache/cassandra/tools/nodetool/ClearSnapshot.java

```diff
@@ -43,34 +46,60 @@ public class ClearSnapshot extends NodeToolCmd
     @Option(title = "clear_all_snapshots", name = "--all", description = "Removes all snapshots")
     private boolean clearAllSnapshots = false;
 
+    @Option(title = "older_than", name = "--older-than", description = "Clear snapshots older than specified time period.")
+    private String olderThan;
+
+    @Option(title = "older_than_timestamp", name = "--older-than-timestamp", description = "Clear snapshots older than specified timestamp.")
+    private Long olderThanTimestamp;
+
     @Override
     public void execute(NodeProbe probe)
     {
-        if(snapshotName.isEmpty() && !clearAllSnapshots)
+        if (snapshotName.isEmpty() && !clearAllSnapshots)
             throw new RuntimeException("Specify snapshot name or --all");
 
-        if(!snapshotName.isEmpty() && clearAllSnapshots)
+        if (!snapshotName.isEmpty() && clearAllSnapshots)
```

Comment:
I think it doesn't make sense to use `--older-than` or `--older-than-timestamp` unless `-all` is also specified. Should we error out if  `--older-than` or `--older-than-timestamp` is specified without `--all` ?

Reply:
@pauloricardomg why is not it a good idea to be able to use it when "--all" is not specified? You might want to remove snapshots only per table or per keyspace and not touch the others.

Reply:
I added test for illegal arguments and their combinations to cover it all.

Btw I noticed that in the current clearsnapshot command we can not clear per table, it was always like that, only per keyspace. But still, we might be interested only in one keyspace and not the others on --all flag so the above logic still holds.